drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -24,20 +24,31 @@ LOG_MODULE_REGISTER(bq274xx, CONFIG_SENSOR_LOG_LEVEL);
 #define BQ274XX_SUBCLASS_DELAY K_MSEC(5)
 
 /* Time to wait for CFGUP bit to be set */
-#define BQ274XX_CFGUP_DELAY K_MSEC(50)
+#define BQ274XX_CFGUP_DELAY K_MSEC(25)
+#define BQ274XX_CFGUP_MAX_TRIES 100
 
 /* Time to set pin in order to exit shutdown mode */
 #define PIN_DELAY_TIME K_MSEC(1)
 
-/* Time it takes device to initialize before doing any configuration */
-#define INIT_TIME K_MSEC(100)
+/* Delay from power up or shutdown exit to chip entering active state, this is
+ * defined as 250ms typical in the datasheet (Power-up communication delay).
+ */
```

Comment:
We should probably be really about which data sheet this was taken from, this info was hard to find.
```suggestion
/* Delay from power up or shutdown exit to chip entering active state, this is
 * defined as 250ms typical in the datasheet (Power-up communication delay).
 */
```
```suggestion
/* Delay from power up or shutdown exit to chip entering active state, this is
 * defined as 250ms typical (Power-up communication delay)
 * see datasheet SLUSBH1C.
 */
```

Reply:
Yeah good idea, linked https://www.ti.com/lit/gpn/BQ27427, it's the most modern one, it has the same line.

Reply:
I could be wrong but have fears this url will break in the future

Reply:
I very much hope that these are meant to be stable, Linux uses them in few spots as well.

Reply:
Okay